### ISSUE TITLE
Execution of issue requests failing with `InvalidScpPledge` error

### ISSUE THREAD

**Issue by ebma** [pendulum-chain/spacewalk#351]
Sometimes the execution of issue requests fails with the following error:
```
vault::issue: Task with slot 46445303 has failed: "SubxtRuntimeError(Runtime(Module(ModuleError { pallet: \"StellarRelay\", error: \"InvalidScpPledge\", description: [], error_data: ModuleErrorData { pallet_index: 68, error: [5, 0, 0, 0] } })))"
```

<img width="1728" alt="image" src="https://github.com/pendulum-chain/spacewalk/assets/6690623/b3a3e586-804d-4caf-b7df-325d811f68cf">

This was encountered for the XLM vault. Restarting the vault also does not fix the issue because it encounters the same error again and again. 

We need to investigate the issue further. Maybe the vault is not sending enough SCP messages. Maybe the SCP messages from the archive are wrong?

**Comment by ebma** [comment]
The problem is very likely that we are not checking if we are only adding `ScpStExternalize` messages when restoring from the archive. We need to add an additional check [here](https://github.com/pendulum-chain/spacewalk/blob/befd6fe8de903407e212902956398df25b50e8f3/clients/vault/src/oracle/collector/proof_builder.rs#L209-L220).

**Comment by ebma** [comment]
While it first seemed like a very simple easy-to-fix problem, it's not. To give a short summary: When the vault client was fetching SCP messages from a Stellar history archive because he missed some messages to build a proof, it did not filter the requested SCP messages for 'externalized' messages but used all of them. Without going into to much detail here, for the transaction verification in the stellar-relay, we are only interested in these externalized messages. Since the vault handed also other unrelated messages to the relay, it threw the error. Now the solution seems simple, we just filter out all other messages that the stellar-relay pallet is not interested in.
During our investigation we however realized that, when querying the SCP messages for a specific slot from the archive node, while 22 to 23 envelopes with SCP messages are returned, just one of these envelopes contains an externalized messages. This is way too little as we need the externalized messages of ~>15 nodes. So the first surprise we encountered here is that the archived SCP messages always only contain exactly 1 externalized message.
Although quite cumbersome, a solution to mitigate this problem would be to fetch not only 1 archive but the Stellar archive of each Stellar validator node and get their externalized message for the target slot.
I tried this and it indeed seems that (although including SCP messages in the history is an optional configuration), all validator node archives do store these messages, which sounds great. But there is a huge but. I decoded the different externalized messages that the archive nodes returned for the slot and it was always the same externalized message. This means that this is completely useless for our purpose because even when fetching from all archive nodes available, apparently we are only able to receive the same message from all of them.

First of, I'd recommend that we increase [this](https://github.com/stellar/stellar-core/blob/ca2fb06059c15442cb4c9a8c89de1a8fc3579a39/docs/stellar-core_example.cfg#L478) parameter even further. We already increased it to 100 as part of [this](https://www.notion.so/satoshipay/Change-stellar-core-and-archive-nodes-with-regard-to-SCP-messages-166d6cd0df1b40a39138168c0c8b12d9) ticket but this is by far the easiest solution for the time being.
This is not perfect though, and we should probably try to find some other solution involving the actual Stellar archive nodes. Maybe we can ask Stellar developers where to add some changes to not only store 1 externalized SCP message per slot in the archive but all of them. Or we ask them why this works like this in the first place and all archives store the same message. Asking someone from Stellar about the details here might be a good idea either way.
